brainscore/metrics/IndividualCeiling.py

```diff
@@ -0,0 +1,56 @@
+from brainscore.metrics.ceiling import Ceiling
+from brainscore.metrics.transformations import Transformation
+from brainscore.benchmarks._neural_common import average_repetition
+import numpy as np
+import xarray as xr
+
+
+class InterIndividualCrossValidation:  # TODO inheritance Transformation???: InterIndivCrossVal = pipe, Ceiling = apply?
+
+    def __call__(self, assembly):  # TODO add support for list of assemblies
+        animals = set(assembly.neuroid.animal.data)  # [set(assembly.neuroid.animal.data) for assembly in assemblies]
+        for animal in animals:
+            relative_complement = xr.concat([assembly.sel(animal=a) for a in animals - {animal}], dim='neuroid')
+            yield relative_complement, assembly.sel(animal=animal)  # assembly.sel(animal=animals - {animal})
+
+
+class InterIndividualBenchmarkCeiling(Ceiling):
+
+    def __init__(self, benchmark, modify_benchmark=lambda benchmark, target: None):
+        self._benchmark = benchmark
+        if modify_benchmark:
+            self._modify_benchmark = modify_benchmark
+        self._set_generator = InterIndividualCrossValidation()
+
+    def __call__(self, assemblies):
+        scores = []
+        for target, source in self._set_generator(assemblies):
+            self._modify_benchmark(self._benchmark, target)  # TODO too hacky??

```

Comment:
Tried to make it generic but seems a little hacky, right?

Reply:
yes I also don't quite understand why you need to modify the benchmark -- wouldn't you be able to apply the metric? I.e. `score = metric(source, target)`.

Reply:
true, I did not think of that...